fmf/utils.py

```diff
@@ -214,15 +214,23 @@ def filter(filter, data, sensitive=True, regexp=False):
     """
     Return true if provided filter matches given dictionary of values
 
-    Filter supports disjunctive normal form with '|' used for OR, '&'
-    for AND and '-' for negation. Individual values are prefixed with
-    'value:', leading/trailing white-space is stripped. For example::
+    Filter supports disjunctive normal (DNF) form with '|' used for OR,
+    '&' for AND and '-' for negation. Individual values are prefixed
+    with 'value:', leading/trailing white-space is stripped. For
+    example:
 
         tag: Tier1 | tag: Tier2 | tag: Tier3
         category: Sanity, Security & tag: -destructive
 
+    Even though parentheses are only allowed in keys or values they are
+    not needed for DNF as the correct order of precedence of operators
+    is respected (negation first, AND next and then OR). For example,
+    to express '(tag: A | tag: B) & tag: C' use the following filter:
```

Comment:
```suggestion
    to express '(tag: A | tag: B) & tag: C' use the following filter::
```

Double colons are need for the online documentation to be properly rendered:
* https://fmf--235.org.readthedocs.build/en/235/modules.html#fmf.filter